hollow/src/main/java/com/netflix/hollow/api/producer/metrics/AnnouncementMetrics.java

```diff
@@ -0,0 +1,64 @@
+package com.netflix.hollow.api.producer.metrics;
+
+import java.util.OptionalLong;
+
+public class AnnouncementMetrics {
+
+    private long dataSizeBytes;                                     // Heap footprint of announced blob in bytes
+    private long announcementDurationMillis;                        // Announcement duration in ms, only applicable to completed cycles (skipped cycles dont announce)
+    private boolean isAnnouncementSuccess;                          // true if announcement was successful, false if announcement failed
+    private OptionalLong lastAnnouncementSuccessTimeNanoOptional;   // monotonic time of last successful announcement (no relation to wall clock), N/A until first successful announcement
+
+
+    public long getDataSizeBytes() {
+        return dataSizeBytes;
+    }
+    public long getAnnouncementDurationMillis() {
+        return announcementDurationMillis;
+    }
+    public boolean getIsAnnouncementSuccess() {
+        return isAnnouncementSuccess;
+    }
+    public OptionalLong getLastAnnouncementSuccessTimeNanoOptional() {
+        return lastAnnouncementSuccessTimeNanoOptional;
+    }
+
+    private AnnouncementMetrics(Builder builder) {
+        this.dataSizeBytes = builder.dataSizeBytes;
+        this.announcementDurationMillis = builder.announcementDurationMillis;
+        this.isAnnouncementSuccess = builder.isAnnouncementSuccess;
+        this.lastAnnouncementSuccessTimeNanoOptional = builder.lastAnnouncementSuccessTimeNanoOptional;
+    }
+
+    public static final class Builder {
+        private long dataSizeBytes;
+        private long announcementDurationMillis;
+        private boolean isAnnouncementSuccess;
+        private OptionalLong lastAnnouncementSuccessTimeNanoOptional;
+
+        public Builder() {
+            lastAnnouncementSuccessTimeNanoOptional = OptionalLong.empty();
+        }
+
+        public Builder setDataSizeBytes(long dataSizeBytes) {
+            this.dataSizeBytes = dataSizeBytes;
+            return this;
+        }
+        public Builder setAnnouncementDurationMillis(long announcementDurationMillis) {
+            this.announcementDurationMillis = announcementDurationMillis;
+            return this;
+        }
+        public Builder setIsAnnouncementSuccess(boolean isAnnouncementSuccess) {
+            this.isAnnouncementSuccess = isAnnouncementSuccess;
+            return this;
+        }
+        public Builder setLastAnnouncementSuccessTimeNanoOptional(OptionalLong lastAnnouncementSuccessTimeNanoOptional) {
```

Comment:
The lack of calling or calling the builder method implies optionally indicating that it does not need to be expressed as a method argument.  A builder does not require that it be built in full fluently with variables accumulated before that being passed as arguments.

Reply:
I think I understand @PaulSandoz's feedback and therefore I think I agree.

Meaning, the argument should just be of type `long` since if we omit the call to `setLastAnnouncementSuccessTimeNanoOptional` then `this.lastAnnouncementSuccessTimeNanoOptional` is already an empty optional.

Reply:
With making the arguments to the builder optional, I'm able to do this in onCycleSkip()-

> cycleMetricsBuilder
>                 .setConsecutiveFailures(consecutiveFailures)
>                 .setIsCycleSuccessOptional(Optional.empty())
>                 .setCycleDurationMillisOptional(OptionalLong.empty())
>                 .setLastCycleSuccessTimeNanoOptional(lastCycleSuccessTimeNanoOptional);
> cycleMetricsReporting(cycleMetricsBuilder.build());

as opposed to this-
> cycleMetricsBuilder
>                 .setConsecutiveFailures(consecutiveFailures)
>                 .setLastCycleSuccessTimeNano(lastCycleSuccessTimeNano);
> cycleMetricsReporting(cycleMetricsBuilder.build());

Both are accurate, but the former makes it more obvious to the reader that we are not reporting the success status or cycle duration for skipped cycles. IMO there is a readability benefit in cases such as this one, where the builder object is not entirely built in one place, so it's easier to follow what we're setting for each parameter.

Could that justify the usage of Optional in builder arguments?

Reply:
> reporting the success status or cycle duration for skipped cycles

Got it. Your rationale makes sense.

Probably worth discussing the `SKIP` status with @PaulSandoz. He wants status to `SUCCESS` or `FAIL` only, for skipped cycles to be successful cycles, and a different means for indicating skipped cycles. I don't know if this PR is the right time for that change. Assuming "no, not the this PR" then I'm fine with the explicit `.empty()` in `onCycleSkip`

Reply:
There are effectively two layers of optionality here. Arguably the declaration with `Optional.isEmpty()` is no more expressive than a code comment (unless you support the reseting of content since you can delete it with no adverse functional consequences). I can make the counterpoint that when `setIsCycleSuccessOptional` is called with a value i have no idea whether it is empty or present without looking at the surrounding code to see how the value is constructed. I think it would be simpler if the builder is specified such that on construction (or after it has been built) it is empty and it can only be accumulated with present items thereby removing the ambiguity. Use of `Optional` is not required in such a model.

FWIW one primary motivation to use a builder is Java's lack of optional/default constructor/method parameters.  If there are only a few ways to construct declare constructors instead of introducing another API artifact.  If it is envisaged that more forms of construction may added later on a builder can be useful to avoid an explosion of telescoping constructors.

Reply:
This suggestion was taken